### PR TITLE
Add delete button to message table

### PR DESCRIPTION
**Pull request by yubiuser**
- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/AdminLTE/blob/master/CONTRIBUTING.md), as well as this entire template.
- [x] I have made only one major change in my proposed changes.
- [x] I have commented my proposed changes within the code.
- [x] I have tested my proposed changes.
- [x] I am willing to help maintain this change if there are issues with it later.
- [x] I give this submission freely and claim no ownership.
- [x] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [x] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
- [x] I have Signed Off all commits. ([`git commit --signoff`](https://git-scm.com/docs/git-commit#Documentation/git-commit.txt---signoff))

---

**What does this PR aim to accomplish?:**

Adds a delete button to the message table

### CODE REVIEW

**Comment by yubiuser** [comment]
I'm stuck here. Despite using `SQLITE3_OPEN_READWRITE` I always get `While executing message statement: attempt to write a readonly database`

Any ideas how to fix this?

File: scripts/pi-hole/js/messages.js [hunk 1]
```diff
@@ -148,3 +164,41 @@ $(function () {
     },
   });
 });
+
+function deleteMessage() {
+  var tr = $(this).closest("tr");
+  var id = tr.attr("data-id");
+
+  utils.disableAll();
+  utils.showAlert("info", "", "Deleting message # ", id);
```
Comment (DL6ER) [inline scripts/pi-hole/js/messages.js]:
```suggestion
  utils.showAlert("info", "", "Deleting message with ID " + parseInt(id, 10), "...");
```

File: scripts/pi-hole/php/message.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+        array_push($response, array('action' => $_POST['action']));
+    }
+    echo json_encode($response);
+}
+
+if ($_POST['action'] == 'delete_message') {
+// Delete message identified by ID
+    try {
+        $db->query('BEGIN TRANSACTION;');
+
```
Comment (DL6ER) [inline scripts/pi-hole/php/message.php]:
```suggestion
```

No need for a transaction as we're doing only one thing.

File: scripts/pi-hole/php/message.php [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            throw new Exception('While binding id to message statement: ' . $db->lastErrorMsg());
+        }
+
+        if (!$stmt->execute()) {
+            throw new Exception('While executing message statement: ' . $db->lastErrorMsg());
+        }
+
+        if(!$db->query('COMMIT;')) {
+            throw new Exception('While commiting changes to the database: ' . $db->lastErrorMsg());
+        }
```
Comment (DL6ER) [inline scripts/pi-hole/php/message.php]:
```suggestion
```

**Review by DL6ER — CHANGES_REQUESTED** [review]

File: scripts/pi-hole/php/message.php [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+{
+    header('Content-type: application/json');
+    $response = array('success' => false, 'message' => $message);
+    if (isset($_POST['action'])) {
+        array_push($response, array('action' => $_POST['action']));
+    }
+    echo json_encode($response);
+}
+
+if ($_POST['action'] == 'delete_message') {
```
Comment (DL6ER) [inline scripts/pi-hole/php/message.php]: I forgot one

```suggestion
if ($_POST['action'] == 'delete_message' && isset($_POST['id'])) {
```

**Comment by DL6ER** [comment]
> Any ideas how to fix this?

Well, the reason is
``` php
$QueriesDB = getQueriesDBFilename();
```
which is `pihole-FTL.db`. However, the web interface of pre-v6.0 has no write permissions to this database file (check the missing group write permission for the `pihole-FTL.db` file):

``` plain
-rw-rw-r-- 1 pihole pihole 8.7M Aug  8 04:41 /etc/pihole/gravity.db
-rw-r--r-- 1 pihole pihole 598M Aug  9 16:00 /etc/pihole/pihole-FTL.db
```

This is, so far, intended and would need a permission change. Note that [FTL itself sets these permissions](https://github.com/pi-hole/FTL/blob/b90ab8b1e78d7aece11c8fee1c3c00817761de52/src/database/common.c#L172-L175) explicitly when creating the database file so the change needs to be done in FTL as well. Let's think about about if this has any bad consequences and, if we agree it does not have any, we can easily add the necessary change to FTL.

**edit** Added link to the relevant FTL code above.

**Comment by yubiuser** [comment]
I see....

What is the risk we are facing? If the web interface can write arbitrary data to the database, we must eliminate the possibility that the data is later interpreted as code but always only displayed as text. 

Messing with existing data should not be a problem - one might loose ones long term history or the network table gets messed up. But nothing should for instance mess with the DNS resolution.

**Comment by DL6ER** [comment]
@yubiuser Did we address the permission issue somewhere? If so, can you link the related PR?

**Comment by yubiuser** [comment]
No, we did not address the permissions so far...

**Comment by yubiuser** [comment]
Follow-up PR https://github.com/pi-hole/AdminLTE/pull/1886

**Pull request closed without merging**